src/plugins/intel_cpu/src/nodes/mha.cpp

```diff
@@ -0,0 +1,1086 @@
+        add(rsp, sizeof(float) * vec_size);
+
+        mov(reg_tmp, dnnl::impl::float2int(1.0f));
+        vmovq(xmm_tmp, reg_tmp);
+        vbroadcastss(get_vmm_denom(0), xmm_tmp);
+        uni_vdivps(get_vmm_denom(0), get_vmm_denom(0), get_vmm_aux(0));
+
+        if (jcp_.with_scales1)
+            mov(reg_scales, ptr[reg_params + GET_OFF(p_scales1)]);
+
+        if (jcp_.with_scales1 && jcp_.broadcast_scales1) {
+            uni_vmovss(Xmm(vmm_scales.getIdx()), ptr[reg_scales]);
+            uni_vbroadcastss(vmm_scales, Xmm(vmm_scales.getIdx()));
+        }
+
+        mov(reg_work_amount_aux, reg_work_amount);
+        L(mul_loop_label);
+        {
+            cmp(reg_work_amount_aux, vec_size);
+            jl(mul_end_label, T_NEAR);
+
+            mul_loop(vec_size);
+
+            sub(reg_work_amount_aux, vec_size);
+
+            jmp(mul_loop_label, T_NEAR);
+        }
+        L(mul_end_label);
+        if (tail_size) {
+            mul_loop(tail_size);
+        }
+
+        this->postamble();
+
+        load_emitter->emit_data();
+        store_emitter->emit_data();
+        exp_emitter->emit_data();
+    }
+
+    void mul_add_max(size_t step) {
+        bool is_tail = step < vec_size;
+
+        load(get_vmm_in(0), reg_mul_in0, jcp_.src_prc, step, is_tail);
+        load(get_vmm_in(2), reg_add_in1, Precision::FP32, step, is_tail);
+
+        if (jcp_.src_prc == Precision::I32) {
+            if (jcp_.with_scales0) {
+                if (!jcp_.broadcast_scales0) {
+                    load(vmm_scales, reg_scales, Precision::FP32, step, is_tail);
+                    add(reg_scales,  sizeof(float) * step);
+                }
+                uni_vmulps(get_vmm_in(0), get_vmm_in(0), vmm_scales);
+                uni_vmaxps(get_vmm_in(0), get_vmm_in(0), vmm_crop_low);
+                uni_vminps(get_vmm_in(0), get_vmm_in(0), vmm_crop_high);
+            }
+
+            uni_vaddps(get_vmm_aux(0), get_vmm_in(0), get_vmm_in(2));
+            uni_vmulps(get_vmm_aux(0), get_vmm_aux(0), get_vmm_in(1));
+        } else {
+            uni_vmulps(get_vmm_aux(0), get_vmm_in(0), get_vmm_in(1));
+            uni_vaddps(get_vmm_aux(0), get_vmm_aux(0), get_vmm_in(2));
+        }
+
+        uni_vmaxps(get_vmm_max(0), get_vmm_max(0), get_vmm_aux(0));
+
+        store(reg_buffer_aux, get_vmm_aux(0), Precision::FP32, step);
+
+        if (!is_tail) {
+            add(reg_mul_in0, jcp_.src_prc.size() * step);
+            add(reg_add_in1, sizeof(float) * step);
+            add(reg_buffer_aux, sizeof(float) * step);
+        }
+    }
+
+    void sub_exp_reduce(size_t step) {
+        bool is_tail = step < vec_size;
+
+        load(get_vmm_in(0), reg_buffer_aux, Precision::FP32, step, is_tail);
+
+        uni_vsubps(get_vmm_in(0), get_vmm_in(0), get_vmm_max(0));
+
+        auto vmm_exp_idx = static_cast<size_t>(get_vmm_in(0).getIdx());
+        exp_emitter->emit_code({vmm_exp_idx}, {vmm_exp_idx}, {}, {});
```

Comment:
As I know, `exp_emitter` need 4 aux vec regs. Also if I miss nothing, we can reuse for that `get_vmm_in(1)`, `get_vmm_in(2)`, `get_vmm_aux(0)` and `get_vmm_aux(1)`

Reply:
Added register pools. Thanks!